src/runtime/disco/nccl/nccl.cc

```diff
@@ -91,26 +90,34 @@ void AllReduce(NDArray send, ReduceKind reduce_kind, NDArray recv) {
   NCCLThreadLocalContext* ctx = NCCLThreadLocalContext::Get();
   ShapeTuple shape = send.Shape();
   int64_t numel = shape->Product();
+  Device device = ctx->worker->default_device;
+  DeviceAPI::Get(device)->SyncStreamFromTo(device, nullptr, ctx->stream);
```

Comment:
one minor comment: please add a field `ctx->compute_stream` and rename `ctx->stream` to `ctx->comm_stream`

Reply:
makes sense